pkg/nodeconfig/nodeconfig.go

```diff
@@ -403,13 +407,51 @@ func (nc *NodeConfig) generateBootstrapKubeconfig(ctx context.Context) (string,
 
 // generateWICDKubeconfig returns the contents of a kubeconfig created from the WICD ServiceAccount
 func (nc *NodeConfig) generateWICDKubeconfig(ctx context.Context) (string, error) {
+	// First, try to use certificate-based authentication if available
```

Comment:
Comment doesn't make much sense unless there is an alternative to cert authentication.